blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/ProofApiUtils.java

```diff
@@ -0,0 +1,32 @@
+package com.sphereon.alfresco.blockchain.agent.proof;
+
+import com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState;
+import com.sphereon.alfresco.blockchain.agent.rest.model.VerifyContentAlfrescoResponse;
+import com.sphereon.sdk.blockchain.proof.model.VerifyContentResponse;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProofApiUtils {
+    public VerifyContentAlfrescoResponse toAlfrescoResponse(final VerifyContentResponse verifyResponse) {
+        final var response = new VerifyContentAlfrescoResponse();
+        response.setRequestId(verifyResponse.getRequestId());
+        response.setBase64Signature(verifyResponse.getBase64Signature());
+        response.setHash(verifyResponse.getHash());
+        response.setHexSignature(verifyResponse.getHexSignature());
+        response.setRegistrationTime(verifyResponse.getRegistrationTime());
+        response.setRegistrationState(this.alfrescoRegistrationStateFrom(verifyResponse.getRegistrationState()));
+        return response;
+    }
+
+    public AlfrescoBlockchainRegistrationState alfrescoRegistrationStateFrom(final VerifyContentResponse.RegistrationStateEnum state) {
+        switch (state) {
+            case NOT_REGISTERED:
+                return AlfrescoBlockchainRegistrationState.NOT_REGISTERED;
```

Comment:
Any reason why the enum doesn't pickup itself on the RegistrationStateEnum value?

Reply:
Yes, that `enum` is in the shared code, which knows about Alfresco and about blockchain. It doesn't know about Proof and therefore the `VerifyContentResponse.RegistrationStateEnum`.